clients/src/main/java/org/apache/kafka/clients/admin/NewPartitionReassignment.java

```diff
@@ -18,16 +18,22 @@
 package org.apache.kafka.clients.admin;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 /**
  * A new partition reassignment, which can be applied via {@link AdminClient#alterPartitionReassignments(Map, AlterPartitionReassignmentsOptions)}.
  */
 public class NewPartitionReassignment {
     private final List<Integer> targetReplicas;
 
+    public static Optional<NewPartitionReassignment> of(Integer... brokers) {
```

Comment:
Does it make sense to change the parameters to `Integer firstBroker, Integer... rest)`? Then assert if `firstBroker` is null. That will make sure that we don't get empty list of brokers and rest of the code that handles this object doesn't have to bother about this case.

Reply:
Could we just check if `brokers == null || brokers.length == 0`?

Reply:
To a client of api its preferable if the contract is visible from the api signature. If we want this restriction, I think better to make it visible in the signature.

Reply:
That makes sense to me. The usage is the same either way - `of(1,2,3)`

Reply:
That's correct. The implementation becomes a bit tricky, as we can't just use `Arrays.asList` anymore.